src/models/Location.java

```diff
@@ -6,97 +6,278 @@
 import java.util.stream.Collectors;
 
 public class Location {
-    private final String name;
-
-    private int x, y;
-    private Guard invariant;
-    private CDD inconsistentPart;
-
-    // Must be final as Automaton expects it to be constant through the lifetime
-    private final boolean isInitial;
-    private boolean isUrgent;
-    private boolean isUniversal;
-    private boolean isInconsistent;
-
-    public Location(String name, Guard invariant, boolean isInitial, boolean isUrgent, boolean isUniversal, boolean isInconsistent, int x, int y) {
+    protected String name;
+    protected int x, y;
+
+    protected Guard invariantGuard;
+    protected CDD invariantCdd;
+
+    protected CDD inconsistentPart;
+
+    protected boolean isInitial;
+    protected boolean isUrgent;
+    protected boolean isUniversal;
+    protected boolean isInconsistent;
+
+    protected List<Location> productOf;
```

Comment:
I initially called the list of locations ``items`` because I called a ``ComplexLocation`` a ``Tuple``. I think ``children`` is fitting but I am a little unsure whether we then have to rename ``isProduct``.

Reply:
I think what `isProduct` tries to capture is to return `true` when the location is a composite of other locations (through automaton operations like the conjunction, parallel product, and quotient operator) and to return `false` when it is a location comming directly from the originally supplied models. Maybe a good name would be `isComposed`. The downside of that name is that people might think this method would only check whether the location is derived from the parallel *composition* operator.

Reply:
If it is ``isComposed`` would ``productOf`` then be renamed to ``composition``, ``elements``, or ``children``?

Reply:
I have changed ``productOf```to ``children``, ``isProduct`` to ``ìsComposition``, and ``createProduct`` to ``createComposition``.